internal/e2e/guest_test.go

```diff
@@ -0,0 +1,81 @@
+/*
+   Copyright 2023 The Kubernetes Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package e2e_test
+
+import (
+	"context"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+
+	wasm "sigs.k8s.io/kube-scheduler-wasm-extension/scheduler/plugin"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/scheduler/test"
+)
+
+// TestGuest_CycleStateCoherence ensures cycle state data is coherent in a
+// scheduling context.
+func TestGuest_CycleStateCoherence(t *testing.T) {
```

Comment:
This is a new test and works because guest/testdata/all/main.go was rewritten to panic on state errors. This is helpful infrastructure because the next state we need to check are cycle state k/v when we add support for that in the next PR.